### PR TITLE
ci: add kubeproxy restart before windows k8s conformance testing

### PR DESCRIPTION
**Pull request by jpayne3506**
**Reason for Change**:

Windows byocni clusters have kubeproxy turned off by default. Ensuring that kubeproxy is restarted before k8s conformance testing occurs ensures that edge cases do not impact windows testing.  

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

File: .pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml [hunk 1]
```diff
@@ -34,6 +34,20 @@ jobs:
             tar -xvzf kubernetes-test-linux-amd64.tar.gz --strip-components=3 kubernetes/test/bin/ginkgo kubernetes/test/bin/e2e.test
 
         displayName: "Setup Environment"
+      - ${{ if contains(parameters.os, 'windows') }}:
+        - script: |
```
Comment (vipul-21) [inline .pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml:38]: Should we add `set -e` incase it fails on any restart cmd ?

**Review by vipul-21 — APPROVED** [review]
